### PR TITLE
VAOS review page

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
This adds the review page for our three variants: direct schedule, va request, and cc request.

There's some missing data because there are a couple outstanding pages and some new fields on one page.

## Testing done
Local and unit testing

## Screenshots
Screen Shot 2019-10-01 at 9 29 11 AM
Screen Shot 2019-10-01 at 9 30 27 AM
Screen Shot 2019-10-01 at 9 31 24 AM

## Acceptance criteria
- [x] Review pages shows up at the end of the flow

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/utils/selectors.js [hunk 1]
```diff
@@ -69,3 +83,14 @@ export function getFacilityPageInfo(state, pageKey) {
       !!formInfo.schema && !!formInfo.schema.properties.vaFacilityMessage,
   };
 }
+
+export function getChosenClinicInfo(state) {
```
Comment (jbalboni) [inline src/applications/vaos/utils/selectors.js]: This is somewhat speculative. I'm assuming we'll store clinics the same way we store facilities, cached according to the data used to fetch them, so we don't have to refetch if someone backs up and changes facilities.

File: src/applications/vaos/containers/ReviewPage.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import ReviewRequestInfo from '../components/ReviewRequestInfo';
+import LoadingButton from 'platform/site-wide/loading-button/LoadingButton';
+
+export class ReviewPage extends React.Component {
+  render() {
+    const { data, facility, clinic } = this.props;
+
+    return (
+      <div>
+        {data.isDirectSchedule && (
```
Comment (jbalboni) [inline src/applications/vaos/containers/ReviewPage.jsx]: I'm assuming we'll set something like this if we hit all the criteria for direct scheduling.

File: src/applications/vaos/newAppointmentFlow.js [hunk 3]
```diff
@@ -76,11 +76,22 @@ export default {
   visitType: {
     url: '/new-appointment/choose-visit-type',
     previous: 'vaFacility',
+    // Update this when reasonForAppointment is merged
     next: 'contactInfo',
   },
+  reasonForAppointment: {
```
Comment (jbalboni) [inline src/applications/vaos/newAppointmentFlow.js]: These two pages don't exist yet, just adding them so I could hook up the Edit links.

**Review by narin — APPROVED** [review]
LGTM

**Review by rianfowler — APPROVED** [review]

**Pull request merged**